File: examples/python/lookup.py
"""
Contoh: cari satu entri Strong's berdasarkan nomornya.
Jalankan: python lookup.py H430
"""

import json
import sys
from pathlib import Path

DICTIONARY_PATH = Path(__file__).resolve().parent.parent.parent / "data" / "combined" / "strongs-full.json"


def load_dictionary() -> list[dict]:
    with DICTIONARY_PATH.open(encoding="utf-8") as f:
        return json.load(f)


def lookup(strong_number: str, dictionary: list[dict]) -> dict | None:
    return next((e for e in dictionary if e["strong_number"] == strong_number), None)


if __name__ == "__main__":
    query = sys.argv[1] if len(sys.argv) > 1 else "H430"
    entry = lookup(query, load_dictionary())

    if entry is None:
        print(f"Nomor Strong '{query}' tidak ditemukan.")
        sys.exit(1)

    print(json.dumps(entry, indent=2, ensure_ascii=False))

File: examples/javascript/lookup.js
// Contoh: cari satu entri Strong's berdasarkan nomornya.
// Jalankan: node lookup.js H430
//
// Jika dipasang lewat npm:
//   npm install strongs-dictionary-json
//   const dictionary = require("strongs-dictionary-json");

const fs = require("fs");
const path = require("path");

const dictionaryPath = path.join(__dirname, "../../data/combined/strongs-full.json");
const dictionary = JSON.parse(fs.readFileSync(dictionaryPath, "utf-8"));

function lookup(strongNumber) {
  return dictionary.find((entry) => entry.strong_number === strongNumber);
}

const query = process.argv[2] || "H430";
const result = lookup(query);

if (!result) {
  console.log(`Nomor Strong '${query}' tidak ditemukan.`);
  process.exit(1);
}

console.log(JSON.stringify(result, null, 2));
